### PR TITLE
Fixes for #2944 and #2945

### PR DESCRIPTION
**Pull request by seriva**
- Refactor InitEngine class to be agnostic to changes in ApplyEngine and UpgradeEngine: #2944
- Removed epicli apply sleeps 10 seconds after creating inventory: #2945
- Rename __ping_retries to ping_retries for consistency.
- Abstract loading of inventory done in a few places to build_io.py
- Minor fixes in dump function.

@przemyslavic Attached are the original init outputs for every provider, full and non-full to compare against:

[any-full.txt](https://github.com/epiphany-platform/epiphany/files/8001903/any-full.txt)
[any-minimal.txt](https://github.com/epiphany-platform/epiphany/files/8001906/any-minimal.txt)
[aws-full.txt](https://github.com/epiphany-platform/epiphany/files/8001907/aws-full.txt)
[aws-minimal.txt](https://github.com/epiphany-platform/epiphany/files/8001908/aws-minimal.txt)
[azure-full.txt](https://github.com/epiphany-platform/epiphany/files/8001909/azure-full.txt)
[azure-minimal.txt](https://github.com/epiphany-platform/epiphany/files/8001910/azure-minimal.txt)

Note that the any-full.txt will be slightly different because its also merged now, rest should be all the same.

### CODE REVIEW

**Review by cicharka — DISMISSED** [review]

File: cli/helpers/build_io.py [hunk 1]
```diff
@@ -53,6 +56,11 @@ def save_inventory(inventory, cluster_model, build_dir=None):
     save_to_file(file_path, content)
 
 
+def load_inventory(inventory_path):
+    inventory = InventoryManager(loader=DataLoader(), sources=inventory_path)
```
Comment (sbbroot) [inline cli/helpers/build_io.py]: Just return InventoryManager in this line, no need for variable.

File: cli/engine/ApplyEngine.py [hunk 2]
```diff
@@ -39,7 +36,7 @@ def __init__(self, input_data):
         self.infrastructure_docs = []
         self.manifest_docs = []
 
-        self.__ping_retries: int = input_data.ping_retries
+        self.ping_retries: int = input_data.ping_retries
```
Comment (sbbroot) [inline cli/engine/ApplyEngine.py:39]: Is there any reason for this variable to be public? I don't see any direct usage of it.

Comment (seriva) [reply cli/engine/ApplyEngine.py:39]: I would agree with you, however the __ paradigm is currently not used  inside the CLI code. So for me this is currently just about consistency. New stuff like the dependency code you are working on or PoC work we should use it and when time allows add it in bulk to the current CLI code as well.

**Review by sbbroot — APPROVED** [review]

**Review by cicharka — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

**Pull request merged**